packages/debug/src/browser/debug-session-manager.ts

```diff
@@ -221,17 +235,31 @@ export class DebugSessionManager {
     }
 
     protected configurationIds = new Map<string, number>();
-    protected async resolveConfiguration(options: Readonly<DebugSessionOptions>): Promise<InternalDebugSessionOptions> {
+    protected async resolveConfiguration(
+        options: Readonly<DebugSessionOptions>
+    ): ResultPromise<InternalDebugSessionOptions> {
```

Comment:
Those methods are implementations. As long as the interface to implement defines return types as `ProviderResult<T>` which really means `T | undefined | null | Thenable<T | undefined  | null>` we can retype implementations to something more specific/convenient. The interface says "one of those can be returned" while the implementation will only return a subset of that.

It's fine to type as `Promise<InternalDebugSessionOptions | undefined>` here.